src/hidimstat/base_variable_importance.py

```diff
@@ -131,3 +134,146 @@ def _check_importance(self):
             raise ValueError(
                 "The importances need to be called before calling this method"
             )
+
+
+class GroupVariableImportanceMixin:
+    """
+    Base class for variable importance methods using feature groups.
+    This class extends `BaseVariableImportance` to support variable importance
+    methods that operate on groups of features, enabling group-wise selection
+    and importance evaluation.
+
+    Attributes
+    ----------
+    n_features_groups : int, default=None
+        The number of feature groups.
+    features_groups : dict, default=None
+        A dictionary mapping group names or indices to lists of feature indices or names.
+    _features_groups_ids : array-like of shape (n_features_groups,), default=None
+        Internal representation of group indices for each group.
+
+    Methods
+    -------
+    fit(X, y=None, groups=None)
+        Identifies and stores feature groups based on input or provided grouping.
+    _check_fit(X)
+        Checks if the class has been fitted and validates group-feature correspondence.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.n_features_groups = None
+        self.features_groups = None
+        self._features_groups_ids = None
```

Comment:
Why not follow the sklearn fitted attributes style and give these attributes an underscore at the end? I guess it would simplify the check_fit and increase compatibility with sklearn

Reply:
This is because there are parameters set in the `fit`.
I for the homogeneity, I will put them to the __init__ as @bthirion ask.

Reply:
Yes exactly, attributes that are set in the fit should be named with an underscore at the end:  `self.n_features_groups` --> `self.n_features_groups_`  and set to None in the __init__